Verify that email addresses are valid before saving request documents

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the code to verify that email addresses are valid before saving request documents.

## 💭 Motivation and context ##

Resolves #94.

## 🧪 Testing ##

I deployed this change to `database1.prod-a.cyhy`, exported a customer with `cyhy-export`, then tinkered with the email addresses and attempted to import said customer via `cyhy-import`.  WHen I used an valid email address the import was successful and when I used an invalid email address the import failed with an error similar to the following:
```console
$ cyhy-import --force REDACTED.json
Traceback (most recent call last):
  File "/usr/local/bin/cyhy-import", line 186, in <module>
    main()
  File "/usr/local/bin/cyhy-import", line 177, in main
    success = import_file(db, args["FILE"], args["--force"], args["--init-stage"])
  File "/usr/local/bin/cyhy-import", line 158, in import_file
    return import_request(db, request, filename, force, init_stage)
  File "/usr/local/bin/cyhy-import", line 145, in import_request
    doc.save()
  File "/usr/local/lib/python2.7/dist-packages/cyhy/db/database.py", line 1310, in save
    raise ValueError(email_address + " is not a valid email address")
ValueError: redacted@@redacted.com is not a valid email address
```

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Deploy this change to `database1.prod-a.cyhy`.

jsf9k committed:
Verify that email addresses are valid when saving request documents

dav3r reviewed (approved):
Excellent! 👍

jsf9k commented:
I'll wait for @mcdonnnj's review before merging this one.

mcdonnnj commented on cyhy/db/database.py:
```diff
@@ -1288,9 +1292,23 @@ def save(self, *args, **kwargs):
+
+        # Verify that all email address fields in fact
+        # contain valid email addresses.
+        contacts = self["agency"].get("contacts")
+        if contacts is not None:
+            for contact in contacts:
+                email_address = contact.get("email")
+                if email_address is not None:
+                    # This should be modified to use fullmatch() when
+                    # this code is ported to Python 3.
+                    if not self.__EmailAddressRegex.match(email_address):
+                        raise ValueError(email_address + " is not a valid email address")
```
Since we're not doing anything special with the default return of `None` I think this would be cleaner.

```suggestion
        contacts = self["agency"].get("contacts", [])
        for contact in contacts:
            email_address = contact.get("email")
            if email_address is not None:
                # This should be modified to use fullmatch() when this code is
                # ported to Python 3.
                if not self.__EmailAddressRegex.match(email_address):
                    raise ValueError(email_address + " is not a valid email address")
```

mcdonnnj reviewed (approved):
LGTM with one suggestion for your consideration.

jsf9k committed:
Make code cleaner

Overriding the default of None when calling get("contacts") allows us to remove one line of code.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
